### DISCUSSION TITLE
ZwCreateFile and CloseFile are not called in pairs

### DISCUSSION THREAD

**Discussion by forworldm** [dokan-dev/dokany#1121] [Q&A]
is this expected behaviour?

**Comment by Liryna** [comment]
You should always have ZwCreateFile followed by a CloseFile but the elapse time depends on when the source application call Close.

**Reply by forworldm** [reply]
the `CloseFile` is not called on some files opened by the windows explorer. even `DokanCloseHandle` doesn't seem to call `CloseFile`.

**Reply by Liryna** [reply]
Explorer will some time keep a handle open while a folder is displayed. You cannot control for how long applications will keep a handle open.
Dokanclosehandle does not close application handles. See the documentation.
